### PR TITLE
Fix loggamma to return inf at x=0 matching scipy

### LINKED ISSUE #38240: jax.scipy.special.loggamma returns nan instead of inf at x=0.0 and x=-0.0
### Description

`jax.scipy.special.loggamma(0.0)` and `jax.scipy.special.loggamma(-0.0)`
return `nan` instead of `inf`.

At `x=0`, the gamma function has a simple pole: `Γ(x) → +∞` as `x → 0`,
so `log(Γ(0)) = +∞`. SciPy correctly returns `inf`. JAX returns `nan`.

Note: negative integer poles (`x = -1, -2, -3, ...`) are handled
consistently by JAX and SciPy (both return `nan`), so the bug is
specific to `x=0`.

## Origin

Found via testing derived from:

```
jax/tests/lax_scipy_special_functions_test.py
  ::LaxScipySpecialFunctionsTest::testScipySpecialFun_loggamma
```

The existing test uses `rand_positive` inputs only, which never reach
`x=0`. The property uses a **reference oracle**: `jax.scipy.special.loggamma(x)`
must agree with `scipy.special.loggamma(x)` in NaN/inf classification
for all inputs.

## Verification (SciPy is correct)

`Γ(x) → +∞` as `x → 0`, so `log(Γ(x)) → +∞`. Confirmed independently
via **mpmath** (arbitrary-precision library, unrelated to SciPy):

```python
import mpmath
for eps in [1e-1, 1e-2, 1e-4, 1e-8, 1e-15]:
    print(f"mpmath.loggamma({eps:.0e}) = {float(mpmath.re(mpmath.loggamma(eps))):.4f}")
# mpmath.loggamma(1e-01) =  2.2527
# mpmath.loggamma(1e-02) =  4.5995
# mpmath.loggamma(1e-04) =  9.2103
# mpmath.loggamma(1e-08) = 18.4207
# mpmath.loggamma(1e-15) = 34.5388
```

The sequence diverges to `+∞`. SciPy is correct; the bug is in JAX.

## Minimal reproducer

```python
import numpy as np
import scipy.special
import jax
import jax.scipy.special

jax.config.update("jax_enable_x64", True)

for x in [np.float32(0.0), np.float32(-0.0)]:
    print(f"x = {x}")
    print(f"  scipy: {scipy.special.loggamma(x)}")
    print(f"  jax:   {jax.scipy.special.loggamma(x)}")
    print(f"  jit:   {jax.jit(jax.scipy.special.loggamma)(jax.numpy.array(x))}")
```

## Expected behavior

```
x = 0.0
  scipy: inf
  jax:   inf
  jit:   inf
x = -0.0
  scipy: inf
  jax:   inf
  jit:   inf
```

## Actual behavior

```
x = 0.0
  scipy: inf
  jax:   nan
  jit:   nan
x = -0.0
  scipy: inf
  jax:   nan
  jit:   nan
```

## Additional cases

```
x= 0.0:  scipy=inf, jax=nan  ← BUG
x=-0.0:  scipy=inf, jax=nan  ← BUG
x=-1.0:  scipy=nan, jax=nan  OK
x=-2.0:  scipy=nan, jax=nan  OK
x=-0.5:  scipy=nan, jax=nan  OK
x= inf:  scipy=inf, jax=inf  OK
x= nan:  scipy=nan, jax=nan  OK
x=-inf:  scipy=nan, jax=nan  OK
```

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:     0.10.1
jaxlib:  0.10.1
numpy:   2.4.6
scipy:   1.17.1
python:  3.12.3
OS:      Linux-6.17.0-29-generic-x86_64-with-glibc2.39
backend: cpu
```

### PR DESCRIPTION
**Pull request by adityasingh2400**
The gamma function has a simple pole at x equals 0, so log of gamma at 0 is positive infinity. SciPy returns inf there, but jax.scipy.special.loggamma returned nan because the real branch kept results only for x greater than 0 and mapped every other value, including 0, to nan.

This returns positive infinity at x equals 0 and at negative zero, while leaving negative reals at nan. That still matches scipy for real-valued inputs, since loggamma is complex-valued on the negative real axis and scipy returns nan for real inputs there. Negative integer poles continue to return nan as before, consistent with scipy.

Verified against the scipy oracle across positive, zero, and negative inputs, confirmed the gradient at a positive point still equals digamma, and the existing loggamma tests in tests/lax_scipy_special_functions_test.py pass (12 passed).

Fixes #38240

### CODE REVIEW

File: jax/_src/scipy/special.py [hunk 1]
```diff
@@ -246,7 +246,12 @@ def loggamma(x: ArrayLike) -> Array:
   if dtypes.issubdtype(x.dtype, np.complexfloating):
     return _complex_loggamma_scipy(x)
   res = lax.lgamma(x)
-  return jnp.where(x > 0, res, jnp.nan)
```
Comment (jakevdp) [inline jax/_src/scipy/special.py:249]: I think the goal here could be accomplished by simply changing this line to
```python
return jnp.where(x >= 0, res, jnp.nan)
```

**Review by jakevdp — CHANGES_REQUESTED** [review]

**Comment by adityasingh2400** [comment]
Thanks @jakevdp, that is cleaner. Done: I dropped the explicit `x == 0` special case and `_lax_const` and now return `jnp.where(x >= 0, res, jnp.nan)`. Since `lax.lgamma` already evaluates to +inf at both 0.0 and -0.0, routing x = 0 through `res` gives the same values and, as the bot noted, keeps the correct nan gradient at the pole instead of zeroing it. Verified locally: loggamma(0.0) and loggamma(-0.0) are inf, negatives are nan, loggamma(3.0) is log(2), and grad at 0 is nan.

**Review by jakevdp — CHANGES_REQUESTED** [review]
Looks good.

Last thing: could we add a regression test that covers this value? It could be as simple as using `jtu.rand_default` rather than `jtu.rand_positive` in the `op_record` for `"loggamma"` in `lax_scipy_special_functions_test.py`.

**Comment by adityasingh2400** [comment]
Thanks for the review. I added a regression test, testLoggammaZeroPole, that covers x=0. The parameterized loggamma case uses rand_positive (and runs grad checks, where the pole gives a nan gradient), so widening that sampler would have broken the autodiff check. Instead I added an explicit test asserting loggamma(0.0) and loggamma(-0.0) return +inf and match scipy.special.loggamma. It passes locally.

**Comment by jakevdp** [comment]
This falls under #38651; please note that it may take a while for this to receive attention from the team.
